services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/util/DeviceRegistryUtils.java

```diff
@@ -184,13 +187,13 @@ public static X509Certificate getCertificateFromClientContext(
                         String.format("Subject DN of the client certificate does not match authId [%s] for tenant [%s]",
                                 authId, tenantId));
             }
-            return cert;
+            return Future.succeededFuture(Optional.of(cert));
         } catch (final CertificateException | ClassCastException | IllegalArgumentException error) {
             final String errorMessage = String.format(
                     "Error getting certificate from client context with authId [%s] for tenant [%s]", authId, tenantId);
             LOG.error(errorMessage, error);
             TracingHelper.logError(span, errorMessage, error);
-            throw new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, error.getMessage());
+            return Future.succeededFuture(Optional.empty());
```

Comment:
this means that the adapter supports auto provisioning but the data is malformed. Thus, this should result in a failed future with a `ClientErrorException(400)`, right?

Reply:
I have fixed now this utility method for getting a certificate from the client context according to your comments and pushed the commit.